Feature add models - Draft

megangilpin opened the pull request:
Created User, Chef and Meal models

Key Questions:
- How to handle availability?
- Do we want validation in the models?

![Screen Shot 2020-11-10 at 5 03 58 PM](https://user-images.githubusercontent.com/48898520/98739339-25f7a880-2377-11eb-8bac-2d206fedbe0f.png)

megangilpin committed:
added models for User, Chef and Meals

megangilpin committed:
fixed postalCode

ThinkTink commented on server/models/chef.js:
```diff
@@ -0,0 +1,25 @@
+    tagLine: { type: Number, required: true },
+    fullBio: { type: Number, required: true },
+  },
+  cuisineSpecialty: { type: Number, required: true },
```
Why is this a number? Also, are they limited to just one?

megangilpin replied:
That should definitely be a string, thanks for catching.  I had limited it to one because that's how I felt the mock up looked but I think its fine if they have more than.

ThinkTink commented on server/models/meal.js:
```diff
@@ -0,0 +1,21 @@
+  servingType: String,
+  cuisineType: [{ type: String, required: true }],
+  ingredients: { type: String, required: true },
+  createdAt: { type: Date, default: Date.now },
```
Your diagram also includes `requirements`, which I think would be good to include

megangilpin replied:
Good catch.  I missed adding that.

ThinkTink commented on server/models/meal.js:
```diff
@@ -0,0 +1,21 @@
+  ingredients: { type: String, required: true },
+  createdAt: { type: Date, default: Date.now },
+  updatedAt: Date,
+  chefId: {type: Schema.Types.ObjectId, ref: 'Chefs'}
```
I'm thinking that we probably only need either the meals array on the Chef model, or the chefId on the Meal model, but not both?
Is there any possibility that two chefs would have the same meal? (I'm thinking not)

megangilpin replied:
I think for now we can just do the chefid on the meal model. I would suppose they could have the same meal or even title.

ThinkTink commented on server/models/user.js:
```diff
@@ -1,11 +1,40 @@
+  first_name: { type: String, required: true },
+  last_name: String,
+  password: { type: Number, required: true },
+  // Do we need reset ability yet?
+  resetToke: Number,
+  resetTokenExpiry: Date,
```
I think we can leave this out for now

megangilpin replied:
No problem!

gary-bains replied:
I'm nit picking here, but the combination of snake_case `first_name` and camelCase `resetToke` bug me

megangilpin replied:
Agree!! Fixed to be all camelCase

ThinkTink commented on server/models/user.js:
```diff
@@ -1,11 +1,40 @@
+  // Do we need reset ability yet?
+  resetToke: Number,
+  resetTokenExpiry: Date,
+  // To do: add validation
```
I think we can do validation in the route that creates the users rather than on the model itself

megangilpin replied:
Great, that will probably be easier especially if we need to make updates to the type of validations.

ThinkTink commented on server/models/user.js:
```diff
@@ -1,11 +1,40 @@
+  bio: String,
+  favoriteCuisine: [{
+    type:String,
+    length: 5,
```
is this a required length or just a maximum?

megangilpin replied:
This was supposed to be the min length of the array but after relooking this is not how you do it.  I don't think we need to worry about since we are doing validation outside of the schema. I had put a min length because if the user picks a ton of foods they don't really have a favorite any more.

ThinkTink reviewed (approved):
I left a couple of comments/questions, other than those I think this is good to merge. We can figure out how to handle availability at a future date

megangilpin committed:
fixed errors and added seed

megangilpin commented:
1. I have updated all the files based off Carmen's comments.
2. Added a script and seed file to add seeds to our local databases if we want it.  This was also to make sure the models were working correctly.

megangilpin committed:
fixed camel case

ThinkTink commented on server/scripts/seedDB.js:
```diff
@@ -0,0 +1,90 @@
+// Seed for users collection
+const userSeed = 
+  {
+    first_name: "Megan",
+    last_name: "Gilpin",
```
Need to match this to the new camelCase in the model.
Can you also add a brief explanation of how to use this to the readme?

megangilpin replied:
Thanks for camelCase call out and will for sure add to the readme.

megangilpin committed:
fixed camelCase and added README

sina-jamshidi commented on server/models/user.js:
```diff
@@ -1,9 +1,30 @@
+  favoriteCuisine: [{ type: String }],
+  allergies: String,
+  isChef: { type: Boolean, default: false },
+  createdAt: { type: Date, default: Date.now },
+  updatedAt: Date
```
I think these are unnecessary for our uses

sina-jamshidi commented on server/models/chef.js:
```diff
@@ -0,0 +1,23 @@
+
+const chefSchema = new Schema({
+  profilePicURL: String,
+  chefBio: {
+    tagLine: { type: String, required: true },
+    fullBio: { type: String, required: true },
+  },
```
I'm wondering about your thoughts on this, but I'm thinking once someone has signed up to be a chef, they no longer need their regular profile and it can be overwritten with their chef profile. In that case we could just have tagLine and Bio in the user model and remove it from here.

megangilpin replied:
Yes, I debated that myself and am fine with your suggestion.  If we do that I am not sure we need the tag line.  We can just use the first line of the bio for the chef's tag line. I think the same can be said for the profile picture.

sina-jamshidi commented on server/models/chef.js:
```diff
@@ -0,0 +1,23 @@
+    date: {type: Date},
+    time: {type: Number}
+  },
+  createdAt: { type: Date, default: Date.now },
+  updatedAt: Date,
```
I think these might be unnecessary as well

sina-jamshidi reviewed (approved):
Just a couple of comments but otherwise looks good to me!

megangilpin committed:
updates for easy sign in

megangilpin committed:
updates for easy sign in

megangilpin committed:
fixed postalCode
